### PR TITLE
Remove obsolete groups

### LINKED ISSUE #8697: Delete obsolete access groups
I went through the access groups and there are quite a few which are no longer in use:

- `Developers`: Looks like Marketplace cruft.
- `Developers Credits`: Credits page is gone.
- `Engagement`: An old group that allowed to export email addresses on an admin page, which was not referenced anywhere, introduced in https://bugzilla.mozilla.org/show_bug.cgi?id=797920.
- `Monolith API`: Not sure what this was about, it's not referenced anywhere. Marketplace?
- `OAuth Partner: Flightdeck`: Not sure what this was about, it's not referenced anywhere. Marketplace?
- `Other Contributors Credits`: Credits page is gone.
- `Past Developers Credits`: Credits page is gone.
- `Persona Reviewer MOTD`: The permission this group uses is not defined (and thus not used) in the code.
- `QA`: Unused group, might have been used at some point for some data dump script that seems to be lost now.
- `Revenue Stats Viewers`: Not referenced anywhere. Marketplace?
- `System Administrators`: Description says "Preserved through permissions migration for future use.". It was created in 2012, is not used in the code and is empty. Remove?

@jvillalobos, @diox, @eviljeff for feedback. Thanks!

### PR DESCRIPTION
**Pull request by wagnerand**
Fix #8697.

### CODE REVIEW

File: src/olympia/migrations/1030-remove-obsolete-groups.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+SET @GROUPS_TO_REMOVE = "Developers,Developers Credits,Engagement,Monolith API,OAuth Partner: Flightdeck,Other Contributors Credits,Past Developers Credits,Persona Reviewer MOTD,QA,Revenue Stats Viewers,System Administrators";
+
+DELETE FROM groups_users WHERE group_id IN (SELECT id FROM groups WHERE FIND_IN_SET(name, @GROUPS_TO_REMOVE));
+DELETE FROM groups WHERE FIND_IN_SET(name, @GROUPS_TO_REMOVE);
```
Comment (diox) [inline src/olympia/migrations/1030-remove-obsolete-groups.sql]: I'd prefer an explicit & simple list of `DELETE FROM groups WHERE name = ""';` statements than this. The list is long enough that it's hard to parse at a glance.

File: src/olympia/migrations/1030-remove-obsolete-groups.sql [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+DELETE FROM groups_users WHERE group_id IN (SELECT id FROM groups WHERE name = 'Monolith API');
+DELETE FROM groups_users WHERE group_id IN (SELECT id FROM groups WHERE name = 'OAuth Partner: Flightdeck');
+DELETE FROM groups_users WHERE group_id IN (SELECT id FROM groups WHERE name = 'Other Contributors Credits');
+DELETE FROM groups_users WHERE group_id IN (SELECT id FROM groups WHERE name = 'Past Developers Credits');
+DELETE FROM groups_users WHERE group_id IN (SELECT id FROM groups WHERE name = 'Persona Reviewer MOTD');
+DELETE FROM groups_users WHERE group_id IN (SELECT id FROM groups WHERE name = 'QA');
+DELETE FROM groups_users WHERE group_id IN (SELECT id FROM groups WHERE name = 'Revenue Stats Viewers');
+DELETE FROM groups_users WHERE group_id IN (SELECT id FROM groups WHERE name = 'System Administrators');
+
+DELETE FROM groups WHERE name = 'Developers');
```
Comment (diox) [inline src/olympia/migrations/1030-remove-obsolete-groups.sql]: Extra parenthesis here and every line below.

Comment (wagnerand) [reply src/olympia/migrations/1030-remove-obsolete-groups.sql]: Uhm, copy-paste, sorry.
